### PR TITLE
Added locator tester modal

### PR DESCRIPTION
**Pull request by dpgraham**
* Added icon to the top bar that opens the modal
* Can select a locator strategy from a hard-coded select dropdown
* Can enter a locator value in an input text
* Shows the results in the same modal, inside a multi-select (only shows the ID's)
* Added a button that clicks on the selected element

screen shot 2017-07-04 at 4 16 11 pm

screen shot 2017-07-04 at 4 11 48 pm

screen shot 2017-07-04 at 4 12 36 pm

screen shot 2017-07-04 at 4 15 40 pm

### CODE REVIEW

**Comment by dpgraham** [comment]
PR is ready now.

'sendKeys' doesn't do anything right now, that will be for the next PR

File: app/renderer/actions/Inspector.js [hunk 1]
```diff
@@ -259,3 +266,47 @@ export function setSessionDetails (sessionDetails) {
...
+export function setLocatorTestStrategy (locatorTestStrategy) {
+  return (dispatch) => {
+    dispatch({type: SET_LOCATOR_TEST_STRATEGY, locatorTestStrategy});
+  };
+}
+
+export function searchForElement (locatorTestStrategy, locatorTestValue) {
+  return async (dispatch) => {
+    dispatch({type: SEARCHING_FOR_ELEMENTS});
+    const {result:elements} = await callClientMethod('elements', [locatorTestStrategy, locatorTestValue]);
```
Comment (jlipps) [inline app/renderer/actions/Inspector.js]: `{result: elements}`

File: app/renderer/components/Inspector/LocatorTestModal.js [hunk 2]
```diff
@@ -0,0 +1,104 @@
...
+      ['-ios predicate string', 'Predicate String (iOS)'],
+      ['-ios class chain', 'Class Chain (iOS)'],
+    ];
+
+    return <Modal visible={isLocatorTestModalVisible} 
+      okText={locatedElements ? 'Done' : 'Search'}
+      cancelText='Cancel'
+      title='Search for element'
+      confirmLoading={isSearchingForElements}
+      onOk={(() => this.onSubmit()).bind(this)}
```
Comment (jlipps) [inline app/renderer/components/Inspector/LocatorTestModal.js]: curious why we need `bind` here? i thought `=>` preserved lexical scope?

File: app/renderer/components/Inspector/LocatorTestModal.js [hunk 3]
```diff
@@ -0,0 +1,104 @@
...
+      onCancel={(() => this.onCancel()).bind(this)}>
+        {!locatedElements && <Row>
```
Comment (jlipps) [inline app/renderer/components/Inspector/LocatorTestModal.js]: we've got a few different sections here, each annotated with `locatedElements` or `!locatedElements`. it's fine but the pattern suggests maybe these are actually two different components?

File: app/renderer/lib/client-frameworks/ruby.js [hunk 4]
```diff
@@ -55,6 +55,10 @@ driver.quit`;
   codeFor_back () {
     return `driver.back`;
   }
+
+  codeFor_clickElement () {
```
Comment (jlipps) [inline app/renderer/lib/client-frameworks/ruby.js]: yeah this is ok for now, but probably we don't want to need to handle `clickElement` when we already handle `click`

File: app/renderer/components/Inspector/LocatorTestModal.js (hunk 2 above)
Comment (dpgraham) [inline app/renderer/components/Inspector/LocatorTestModal.js]: The method handlers bind `this` to `event`

Comment (jlipps) [inline app/renderer/components/Inspector/LocatorTestModal.js]: hmm, still seems wonky. would be nice if we could do one of these other things:

```js
onOk={this.onSubmit}
```

```js
onOk={() => this.onSubmit()}
```

```js
onOk={this.onSubmit.bind(this)} // if binding is necessary
```

**Comment by jlipps** [comment]
also, do you know why the CI failed?

**Comment by dpgraham** [comment]
It always fails due to code-signing.

File: app/renderer/lib/client-frameworks/ruby.js (hunk 4 above)
Comment (dpgraham) [inline app/renderer/lib/client-frameworks/ruby.js]: This will be gone in a future PR.

**Comment by dpgraham** [comment]
Fixes made.

The next PR will have the recorder updates. I just wanted to split them into two PR's because it would be pretty tough to review them all at once.

**Comment by Ysoko** [comment]
Any update as to when this will be publicly available? Updated to Appium Desktop 1.1.1 today and still do not see this feature, very interested in being able to test XPATH's again.

**Pull request merged**